elementary/operations/upload_source_freshness.py

```diff
@@ -42,13 +45,31 @@ def upload_results(self, results: dict):
         if not dbt_project_utils.is_dbt_package_up_to_date():
             dbt_runner.deps()
 
+        invocation_id = metadata.get("invocation_id")
+        if not invocation_id:
+            raise click.ClickException("No invocation id found in sources.json.")
+
+        response = dbt_runner.run_operation(
+            "elementary_cli.can_upload_source_freshness",
+            macro_args={"invocation_id": invocation_id},
+            quiet=True,
+        )
+        if not response:
+            raise click.ClickException(
+                f"Source freshness for invocation id {invocation_id} were already uploaded."
+            )
+
         chunk_size = 100
         chunk_list = list(range(0, len(results), chunk_size))
         upload_with_progress_bar = alive_it(
             chunk_list, title="Uploading source freshness results"
         )
         for chunk in upload_with_progress_bar:
             results_segment = results[chunk : chunk + chunk_size]
+
+            for result in results_segment:
```

Comment:
What's the purpose of this?

Reply:
need to pass the metadata to flatten_source_freshness for each record